### ISSUE TITLE
Lsp-fsharp launch command fails due to `/usr/bin/env --default-signal` wrapper

### ISSUE THREAD

**Issue by jthetzel** [emacs-lsp/lsp-mode#3541]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [ ] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

`lsp-fsharp--make-launch-cmd` wraps `fsautocomplete` with `/usr/bin/env --default-signal` ( https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-fsharp.el#L173 ). However, the `--default-signal` flag is only avalable for coreutils v8.31 and higher ( http://git.savannah.gnu.org/gitweb/?p=coreutils.git;a=log;h=refs/tags/v8.31 ). Ubuntu 20.04 LTS currently provides coreutils v8.30 ( https://ubuntu.pkgs.org/20.04/ubuntu-main-amd64/coreutils_8.30-3ubuntu2_amd64.deb.html ).

### Steps to reproduce

Versions:
```
$ emacs --version

GNU Emacs 29.0.50
Development version f7324b7a6442 on master branch; build date 2022-05-29.
Copyright (C) 2022 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.

$ lsb_release -a

No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal

$ /usr/bin/env --version

env (GNU coreutils) 8.30
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Richard Mlynarik, David MacKenzie, and Assaf Gordon.
```

Fsautocomplete command fails:
```
$ /usr/bin/env --default-signal fsautocomplete --version

/usr/bin/env: unrecognized option '--default-signal'
Try '/usr/bin/env --help' for more information.
```

### Expected behavior

Fsautocomplete command succeeds:
```
$ fsautocomplete --version

0.54.0+b69388590dcc144953e1dcd49ac80efb7c1c6546
```

### Which Language Server did you use?

lsp-fsharp

### OS

Linux

### Error callstack

```emacs-lisp
;; *lsp-log*
Found the following clients for /home/jthetzel/src/c-core-labs/bct/Script.fsx: (server-id fsac, priority 0)
The following clients were selected based on priority: (server-id fsac, priority 0)


;; *fsac::stderr*
/usr/bin/env: unrecognized option '--default-signal'
Try '/usr/bin/env --help' for more information.

Process fsac stderr finished
```

### Anything else?

Depending on the guidance of others, we can provide:
1) a fallback within `lsp-fsharp--make-launch-cmd` if `coreutils` version is less than 0.31, or
2) a helpful error message indicating `coreutils` v0.31 or greater required.

**Comment by yyoncho** [comment]
cc @razzmatazz

**Comment by razzmatazz** [comment]
this workaround is not needed anymore for linux, as the breage is now limited only to emacs-28.1@macos; master and macos-28.2 has the issue fixed;

will do a PR

**Comment by razzmatazz** [comment]
@yyoncho I have upped a PR for this:
- https://github.com/emacs-lsp/lsp-mode/pull/3542

**Comment by razzmatazz** [comment]
@jthetzel this should be fixed as of 8123c5cd33bedf3ac5e88d98ede690f73b7f1b00

**Comment by jthetzel** [comment]
Thanks, @razzmatazz ! Your fix and PR merge are lightning quick and enable me to move forward. Cheers.
